test/mvlognormal.jl

```diff
@@ -92,46 +92,44 @@ end
 
 ####### Validate results for a single-dimension MvLogNormal by comparing with univariate LogNormal
 @testset "Comparing results from MvLogNormal with univariate LogNormal" begin
-l1 = LogNormal(0.1,0.4)
-l2 = MvLogNormal(0.1*ones(1),0.4)
-@test [mean(l1)]     ≈ mean(l2)
-@test [median(l1)]   ≈ median(l2)
-@test [mode(l1)]     ≈ mode(l2)
-@test [var(l1)]      ≈ var(l2)
-@test  entropy(l1)   ≈ entropy(l2)
-@test logpdf(l1,5.0) ≈ logpdf(l2,[5.0])
-@test pdf(l1,5.0)    ≈ pdf(l2,[5.0])
-@test (Random.seed!(78393) ; [rand(l1)]) == (Random.seed!(78393) ; rand(l2))
-@test [rand(MersenneTwister(78393), l1)] == rand(MersenneTwister(78393), l2)
+    l1 = LogNormal(0.1,0.4)
+    l2 = MvLogNormal(0.1*ones(1),0.4)
+    @test [mean(l1)]     ≈ mean(l2)
+    @test [median(l1)]   ≈ median(l2)
+    @test [mode(l1)]     ≈ mode(l2)
+    @test [var(l1)]      ≈ var(l2)
+    @test  entropy(l1)   ≈ entropy(l2)
+    @test logpdf(l1,5.0) ≈ logpdf(l2,[5.0])
+    @test pdf(l1,5.0)    ≈ pdf(l2,[5.0])
+    @test (Random.seed!(78393) ; [rand(l1)]) == (Random.seed!(78393) ; rand(l2))
+    @test [rand(MersenneTwister(78393), l1)] == rand(MersenneTwister(78393), l2)
 end
 
 ###### General Testing
 
-mu = [0.1, 0.2, 0.3]
-va = [0.16, 0.25, 0.36]
-C = [0.4 -0.2 -0.1; -0.2 0.5 -0.1; -0.1 -0.1 0.6]
-
-@testset "Testing MvLogNormal with $key" for (key, rng) in
-    Dict("rand(...)" => missing,
-         "rand(rng, ...)" => MersenneTwister(123))
-    
-    @testset "Testing MvLogNormal with $key for $(typeof(g)) with $(Distributions.distrname(g.normal))" for (g, μ, Σ) in [
+@testset "Lognormal tests" begin
+    mu = [0.1, 0.2, 0.3]
+    mu_r = 0.1:0.1:0.3
+    va = [0.16, 0.25, 0.36]
+    C = [0.4 -0.2 -0.1; -0.2 0.5 -0.1; -0.1 -0.1 0.6]
+    for (g, μ, Σ) in [
         (MvLogNormal(mu,PDMats.PDMat(C)), mu, C),
+        (MvLogNormal(mu_r,PDMats.PDMat(C)), mu_r, C),
         (MvLogNormal(PDMats.PDiagMat(sqrt.(va))), zeros(3), Matrix(Diagonal(va))),
         (MvLogNormal(mu, sqrt(0.2)), mu, Matrix(0.2I, 3, 3)),
         (MvLogNormal(3, sqrt(0.2)), zeros(3), Matrix(0.2I, 3, 3)),
-        (MvLogNormal(mu, sqrt.(va)), mu, Matrix(Diagonal(va))),
-        (MvLogNormal(sqrt.(va)), zeros(3), Matrix(Diagonal(va))),
+        (MvLogNormal(mu, Vector{Float64}(sqrt.(va))), mu, Matrix(Diagonal(va))), # Julia 0.4 loses type information so Vector{Float64} can be dropped when we don't support 0.4
+        (MvLogNormal(Vector{Float64}(sqrt.(va))), zeros(3), Matrix(Diagonal(va))), # Julia 0.4 loses type information so Vector{Float64} can be dropped when we don't support 0.4
         (MvLogNormal(mu, C), mu, C),
         (MvLogNormal(C), zeros(3), C) ]
-        
+
+        println("    testing $(typeof(g)) with normal distribution $(Distributions.distrname(g.normal))")
```

Comment:
I think this could just be another `@testset`.

Reply:
splitting which part? The only test in the loop are the three below this line